pyqg/kernel.pyx

```diff
@@ -344,7 +344,7 @@ cdef class PseudoSpectralKernel:
                     # overwrite the tendency, since the forcing gets called after
                     self.dqhdt[k,j,i] = -( self._ik[i] * self.uqh[k,j,i] +
                                     self._il[j] * self.vqh[k,j,i] +
-                                    self._ikQy[k,i] * self.ph[k,j,i] )
+                                    self._ikQy[k,j,i] * self.ph[k,j,i] )
```

Comment:
Looks good...here the advection of the now *3D* background PV gradient gets added to the tendency.